client/state/jetpack-connect/selectors.js

```diff
@@ -96,3 +97,14 @@ export const hasExpiredSecretError = function( state ) {
 		includes( get( authorizeData, [ 'authorizeError', 'message' ] ), 'verify_secrets_expired' )
 	);
 };
+
+/**
+ * Returns the partner_id query param if present or null.
+ *
+ * @param {object}    state Global state tree
+ * @return {null|int}       The partner ID as an integer or null
+ */
+export const getPartnerIdFromQuery = function( state ) {
```

Comment:
There's a [guideline](https://github.com/Automattic/wp-calypso/blob/master/client/state/selectors/README.md) that new selectors should go into their own file in the `client/state/selectors` directory. I'd like to follow that here, especially since this selector is expected to be used in multiple sections, not just `jetpack-connect`.